File: server/src/main/java/org/kane/domain/DTO/entityDTO/diary/meal/MealCreateDTO.java
package org.kane.domain.DTO.entityDTO.diary.meal;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.kane.database.enum_types.MealType;
import org.kane.domain.DTO.entityDTO.diary.meal_item.MealItemCreateDTO;

import java.time.LocalDate;
import java.time.LocalTime;
import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class MealCreateDTO {
    private LocalDate dailyRecordDate;
    private MealType mealType;
    private LocalTime time;
    private List<MealItemCreateDTO> list;
}

File: client/src/app/DTO/entity_dto/diary/meal/meal-create.dto.ts
import { type CalendarDate, MealType } from '../../../types';
import MealItemCreateDTO from '../meal_item/meal-item-create.dto';

export default interface MealCreateDTO {
  dailyRecordDate: CalendarDate;
  mealType: MealType;
  time: string;
  list: MealItemCreateDTO[];
}
